lib/liquid/filters.ex

```diff
@@ -80,4 +448,24 @@ defmodule Liquid.Filters do
     filter(rest, ret)
   end
 
+  def filter(a, b, _) when a == [] do
+    filter(a, b)
+  end
+
+  def filter([head|tail], b, context) do
+    case context.assigns do
```

Comment:
I'll take a deeper look. Currently `Variable.create` and `lookup` are dealing with unparsed values. I think that private `resolve` function can deal with that matching.
Was thinking of variable matching even before passing them to filters actually. Does it make sense?